src/App.tsx

```diff
@@ -1,26 +1,249 @@
+
+      return updated;
+    });
+
+    addTodo(inputText.trim())
+      .then(newTodo => {
+        setTodos(prevTodos => [...prevTodos, newTodo]);
+        setTempTodos(prevTempTodos =>
+          prevTempTodos.filter(todo => todo.id !== tempTodo.id),
+        );
+        setInputText('');
+        setValidationError('');
+        if (inputRef.current) {
+          inputRef.current.focus(); // Устанавливаем фокус на текстовое поле после успешного добавления
+        }
+      })
+      .catch(() => {
+        setLoadingError('Unable to add a todo');
+        setTempTodos(prevTempTodos =>
+          prevTempTodos.filter(todo => todo.id !== tempTodo.id),
+        );
+        setTimeout(() => {
+          setLoadingError('');
+        }, 3000);
+      })
+      .finally(() => {
+        setLoadingTodos(prev => {
+          const updated = prev.filter(id => id !== tempTodo.id);
+
+          return updated;
+        });
+        if (inputRef.current) {
+          inputRef.current.focus(); // Устанавливаем фокус на текстовое поле в блоке finally
+        }
+      });
+  };
+
+  const deletePost = (id: number) => {
+    setLoadingTodos(prev => [...prev, id]);
+
+    deleteTodo(id)
+      .then(() => {
+        setTodos(prevTodos => prevTodos.filter(todo => todo.id !== id));
+      })
+      .catch(() => {
+        setLoadingError('Unable to delete a todo');
+        setTimeout(() => {
+          setLoadingError('');
+        }, 3000);
+      })
+      .finally(() => {
+        setLoadingTodos(prev => prev.filter(todoId => todoId !== id));
+      });
+  };
+
+  const clearErrors = () => {
+    setLoadingError('');
+    setValidationError('');
+  };
+
+  const clearCompleted = () => {
+    const completedTodoIds = todos
+      .filter(todo => todo.completed)
+      .map(todo => todo.id);
+
+    completedTodoIds.forEach(id => {
+      deletePost(id);
+    });
+  };
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setInputText(e.target.value);
+  };
+
+  const leftItem = todos.filter(todo => !todo.completed).length;
+  const disabledButton = todos.every(todo => !todo.completed);
+
+  const filterValues = Object.values(Filter);
+
+  const allTodos = [...filterTodos, ...tempTodos];
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">
-          React Todo App - Load Todos
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          <button
+            type="button"
+            className={`todoapp__toggle-all ${todos.every(todo => todo.completed) ? 'active' : ''}`}
```

Comment:
Move this logic `todos.every(todo => todo.completed` to the helper variable and use it here. Also, use the classnames library rules for add classes by condition
```suggestion
            className={`todoapp__toggle-all ${todos.every(todo => todo.completed) ? 'active' : ''}`}
```